scipy/optimize/meson.build

```diff
@@ -0,0 +1,191 @@
+  dependencies : [py3_dep],
+  install: true,
+  subdir: 'scipy/optimize')
+
+moduleTNC = py3.extension_module('moduleTNC',
+  ['tnc/tnc.h', 'tnc/moduleTNC.c', 'tnc/tnc.c'],
+  c_args: numpy_nodepr_api,
+  include_directories : [inc_np, inc_f2py],
+  dependencies : [py3_dep],
+  install: true,
+  subdir: 'scipy/optimize')
+
+cobyla_module = custom_target('cobyla_module',
+  output : ['_cobylamodule.c'],
+  input : 'cobyla/cobyla.pyf',
+  command : [py3, generate_f2pymod, '@INPUT@', '-o', '@OUTDIR@'],
+)
+
+_cobyla = py3.extension_module('_cobyla',
+  [cobyla_module, fortranobject_c, 'cobyla/cobyla2.f', 'cobyla/trstlp.f'],
+  c_args: numpy_nodepr_api,
+  fortran_args : fortran_ignore_warnings,
+  include_directories : [inc_np, inc_f2py],
+  dependencies : [py3_dep],
+  install: true,
+  subdir: 'scipy/optimize')
+
+minpack2_module = custom_target('minpack2_module',
+  output : ['minpack2module.c'],
+  input : 'minpack2/minpack2.pyf',
+  command : [py3, generate_f2pymod, '@INPUT@', '-o', '@OUTDIR@'],
+)
+
+minpack2 = py3.extension_module('minpack2',
+  [minpack2_module, fortranobject_c, 'minpack2/dcsrch.f', 'minpack2/dcstep.f'],
+  c_args: numpy_nodepr_api,
+  fortran_args : fortran_ignore_warnings,
+  include_directories : [inc_np, inc_f2py],
+  dependencies : [py3_dep],
+  install: true,
+  subdir: 'scipy/optimize')
+  
+slsqp_module = custom_target('slsqp_module',
+  output : ['_slsqpmodule.c'],
+  input : 'slsqp/slsqp.pyf',
+  command : [py3, generate_f2pymod, '@INPUT@', '-o', '@OUTDIR@'],
+)
+
+_slsqp = py3.extension_module('_slsqp',
+  [slsqp_module, fortranobject_c, 'slsqp/slsqp_optmz.f'],
+  c_args: numpy_nodepr_api,
+  fortran_args : fortran_ignore_warnings,
+  include_directories : [inc_np, inc_f2py],
+  dependencies : [py3_dep],
+  install: true,
+  subdir: 'scipy/optimize')
+
+__nnls_module = custom_target('__nnls_module',
+  output : ['__nnlsmodule.c'],
+  input : '__nnls/nnls.pyf',
+  command : [py3, generate_f2pymod, '@INPUT@', '-o', '@OUTDIR@'],
+)
+
+py3.install_sources(
+    ['__nnls.pyi'],
+    pure: false,
+    subdir: 'scipy/optimize'
+)
+
+__nnls = py3.extension_module('__nnls',
+  ['__nnls/nnls.f', __nnls_module, fortranobject_c],
+  c_args: numpy_nodepr_api,
+  fortran_args : fortran_ignore_warnings,
+  include_directories : [inc_np, inc_f2py],
+  dependencies : [py3_dep],
+  install: true,
+  subdir: 'scipy/optimize')
+
+res = run_command(py3, '-c', 'import os; print(os.environ.get("SCIPY_USE_PYTHRAN", 1))')
+if res.returncode() == 0
+  # DOES THIS WORK?
+  _group_columns = custom_target('_group_columns',
+    output: ['_group_columns.so'],
+    input : '_group_columns.py',
+    command : [pythran, '@INPUT@', '-o', '@OUTDIR@/_group_columns.so'],
+  )
+  _group_columns = py3.extension_module('_group_columns',
+  [_group_columns],
+  install: true,
+  subdir: 'scipy/optimize')
+else
+  _group_columns = py3.extension_module('_group_columns',
+  ['_group_columns.c'],
+  c_args: numpy_nodepr_api,
```

Comment:
You want to leave out this line. Unfortunately Cython generates code that targets the deprecated NumPy API. This will only be fixed when Cython 3.0 comes out (probably next year).